include/common/coredump.h

```diff
@@ -0,0 +1,279 @@
+    const auto *CurrentInstance = StackMgr.getModule();
+
+    auto Core = collectProcessInformation(CurrentInstance);
+    // auto DataSection = collectDataSection(StackMgr);
+    auto MemSec = collectMemory(StackMgr);
+    // auto Globals = collectGlobals(StackMgr);
+    auto CoreStack = collectCoreStack(StackMgr);
+    // auto CoreInstances = collectCoreInstances(StackMgr);
+    // auto CoreModules = collectCoreModules(StackMgr);
+
+    //  final file generation
+    AST::Module Mod{};
+    std::vector<Byte> &Magic = Mod.getMagic();
+    Magic.insert(Magic.begin(), {0x00, 0x61, 0x73, 0x6d});
+
+    std::vector<Byte> &Version = Mod.getVersion();
+    Version.insert(Version.begin(), {0x01, 0x00, 0x00, 0x00});
+
+    Mod.getCustomSections().emplace_back(Core);
+    // Mod.getCustomSections().emplace_back(CoreModules);
+    // Mod.getCustomSections().emplace_back(CoreInstances);
+    Mod.getCustomSections().emplace_back(CoreStack);
+    // Mod.getDataSection() = std::move(DataSection);
+    Mod.getMemorySection() = std::move(MemSec);
+    // TODO globals works with DWARF data, is this section needed?
+    //  Mod.getGlobalSection() = std::move(Globals);
+    auto Res = Ser.serializeModule(Mod);
+    std::ofstream File("coredump.wasm", std::ios::out | std::ios::binary);
+    if (File.is_open()) {
+      File.write(reinterpret_cast<const char *>(Res->data()), Res->size());
+      File.close();
+    } else {
+      // Handle error opening file
+      throw std::ios_base::failure("Failed");
+    }
+  }
+
+  AST::CustomSection collectProcessInformation(
+      const Runtime::Instance::ModuleInstance *CurrentInstance) {
+    spdlog::info("Constructing Core");
+    // AST node
+    AST::CustomSection Core;
+    Core.setName("core");
+
+    // Insert in data and set Name of section
+    auto Name = CurrentInstance->getModuleName();
+    std::vector<Byte> Data(reinterpret_cast<const Byte *>(Name.data()),
+                           reinterpret_cast<const Byte *>(Name.data()) +
+                               Name.size());
+
+    auto &Content = Core.getContent();
+    if (Data.size() != 0) { // case where the module has an empty name field
+      Content.insert(Content.end(), Data.begin(), Data.end());
+    } else {
+      Content.insert(Content.begin(), {0x00, 0x00});
+    }
+
+    return Core;
+  }
+
+  AST::MemorySection collectMemory(Runtime::StackManager &StackMgr) {
+    spdlog::info("Collecting memory section");
+
+    AST::MemorySection MemSec;
+    const auto *CurrentInstance = StackMgr.getModule();
+    auto &MemInstances = CurrentInstance->getOwnedMemoryInstances();
+    // In the current version of WebAssembly, at most one memory may be defined
+    // or imported in a single module, and all constructs implicitly reference
+    // this memory.
+    std::cout << MemInstances[0]->getMemoryType().getLimit().getMin();
+    // auto &Content = MemSec.getContent();
+    // TODO why is this setting max and min to the same value?
```

Comment:
Please read the spec of `Limit` about the meaning of the maximum and minimum values.

Reply:
Regarding this, I think the coredump-rewriter(my reference for coredump) itself maybe gets this wrong. Upon inspecting the memory section of the original binary that generates the trap, I can see that: `0	min=17 max=none` where 0 is the index. For some reason coredump-rewriter sets this to `initial: 0, maximum: Some(17)`